File: Problem122_BestTimetoBuyandSellStock.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit=0
        for i in range(len(prices)-1):
            prices[i] = prices[i+1]-prices[i]
            max_profit += prices[i] if prices[i] > 0 else 0
        return max_profit
    
prices = [1,2,3,4,5]
solve = Solution()
print(solve.maxProfit(prices))

File: Problem122_BestTimetoBuyandSellStock.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit=0;
        for (int i=0;i<prices.size()-1;i++) {
            prices[i] = prices[i+1] - prices[i];
            max_profit += prices[i] > 0 ? prices[i] : 0;
        }
        return max_profit;
    }
};

int main() {
    vector<int> prices = {7,1,5,3,6,4};
    Solution solve;
    cout << solve.maxProfit(prices);
}